OpenRA.Mods.Common/Widgets/Logic/Ingame/ReplayControlBarLogic.cs

```diff
@@ -86,7 +86,7 @@ public ReplayControlBarLogic(Widget widget, World world, OrderManager orderManag
 
 				var maximumButton = widget.Get<ButtonWidget>("BUTTON_MAXIMUM");
 				maximumButton.IsHighlighted = () => speed == PlaybackSpeed.Maximum;
-				maximumButton.IsDisabled = () => orderManager.NetFrameNumber >= replayNetTicks;
+				maximumButton.IsDisabled = () => orderManager.IsOutOfSync || orderManager.NetFrameNumber >= replayNetTicks;
```

Comment:
We have this exact statement 6 times. I suggest adding a local function and using that everywhere:
```cs
bool ShouldDisableCommandButton() => orderManager.IsOutOfSync || orderManager.NetFrameNumber >= replayNetTicks;
```

Actually the Pause button's `IsVisible` could use a `!ShouldDisableCommandButton()`, so that's 7.